modules/c++/six.sicd/include/six/sicd/Utilities.h

```diff
@@ -19,29 +19,140 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <utility>
 
 #include <scene/SceneGeometry.h>
 #include <scene/ProjectionModel.h>
-#include <six/sicd/ComplexData.h>
-
-namespace six
-{
-namespace sicd
-{
-class Utilities
-{
-public:
+#include <six/sicd/ComplexData.h>
+
+#include <six/NITFReadControl.h>
+
+namespace six
+{
+namespace sicd
+{
+class Utilities
+{
+public:
     static scene::SceneGeometry* getSceneGeometry(const ComplexData* data);
-
-    static scene::ProjectionModel* getProjectionModel(const ComplexData* data, 
+
+    static scene::ProjectionModel* getProjectionModel(const ComplexData* data, 
             const scene::SceneGeometry* geom);
 
+   
+    /*
+     * getLoadedNitfReadControl creates and loads a NITFReadControl for the
+     * specified file.
+     *
+     * \param sicdPathname SICD NITF pathname
+     * \param schemaPaths One or more files or directories containing SICD
+     * schemas
+     *
+     * \return a NITFReadControl 'load()'ed with the given SICD
+     *
+     */
+    static
+    NITFReadControl getLoadedNITFReadControl(const std::string& sicdPathname,
+                                 const std::vector<std::string>& schemaPaths);
+
+    /*
+     * SicdContents is a helper object for readSicdContents
+     *
+     * This class is a simple wrapper around dynamically allocated
+     * SICD information designed to allow access to either or both 
+     * variables returned by readSicdContents.
+     *
+     */
+    struct SicdContents {
+        // allocated with new
+        ComplexData * complexData;
+        // allocated with new[]
+        std::complex<float>* widebandData;
+    };
+
+    /*
+     * Given a SICD path name and a list of schema, this function reads
+     * and parses the SICD in order to provide the wideband data as well
+     * as the ComplexData associated with the image..
+     *
+     * \param sicdPathname SICD NITF pathname
+     * \param schemaPaths One or more files or directories containing SICD 
+     * schemas
+     *
+     * \return ComplexData associated with the SICD NITF,
+     *           allocated with new
+     *         std::complex<float> buffer containing SICD wideband data,
+     *           allocated with new[]
+     *
+     * \throws See six::sicd::Utilities::getComplexData and 
+     *           six::sicd::Utilities::getWidebandData
+     *
+     */
+    static SicdContents readSicd(const std::string& sicdPathname,
+                                 const std::vector<std::string>& schemaPaths);
+
+    /*
+     * Given a reference to a loaded NITFReadControl, this function
+     * parses the SICD's XML and returns a new ComplexData object.
+     *
+     * \param reader A NITFReadControl loaded with the desired SICD
+     *
+     * \return ComplexData new ComplexData associated with the SICD NITF
+     *
+     * \throws except::Exception if the provided reader is not a SICD
+     *
+     */
+    static ComplexData* getComplexData(NITFReadControl& reader);
```

Comment:
So here is my rationale on this.  On the happy day that we can move over to C++11, big chunks of mem will themselves be deprecated too (i.e. you could use a `mem::SharedPtr` now, but at the point that we're all-in with C++11, we'll remove it anyway since it's just a poor man's `std::shared_ptr`).  There are two schools of thought... Scott Meyers says to just always use shared pointers since they'll do what you want.  I prefer to clearly express the ownership based on the smart pointer type.  So, by returning an `auto_ptr`, you're making it clear that you're the only owner of this thing (which in this case as the caller you probably don't care but it still adds a little clarity).  When we go to C++11, we'll do a find/replace and change all `std::auto_ptr`s over to the new and improved `std::unique_ptr`.  In the meantime, I turned off the gcc warning (in build.py) about auto_ptr being deprecated since there's nothing we can do about it presently.